lamindb/curators/__init__.py

```diff
@@ -911,6 +918,296 @@ def __init__(
         self._columns_field = self._var_fields
 
 
+class CatColumn:
+    """Categorical column for `DataFrame`.
+
+    Args:
+        values_getter: A callable or iterable that returns the values to validate.
+        field: The field to validate against.
+        key: The name of the column to validate. Only used for logging.
+        values_setter: A callable that sets the values.
+        organism: The organism to validate against.
+        source: The source to validate against.
+    """
+
+    def __init__(
+        self,
+        values_getter: Callable | Iterable[str],
+        field: FieldAttr,
```

Comment:
Why doesn't this take a `dtype` so that one can express things like sub-types or species or ontology versions?

Reply:
New features are not yet implemented, this PR is only refactoring existing code.